### DISCUSSION TITLE
Detecting new audio interface.

### DISCUSSION THREAD

**Discussion by dfoster81** [mcguirepr89/BirdNET-Pi#677] [Q&A] [answered]
Hi All, new user here, very exited with the project. Thanks so much for pulling it together. I believe I've searched similar topics, and have also hit reddit with not much to go on there, so please don't flame me. 

TLDR: Do I need to restart pulse/birdnet to detect an audio interface that initializes after bootup?

Bit of a background, I'm in a fairly standard urban area with traffic noise. Home setup is just an Elgato Wave XLR with a cheap condenser mic, MXL 770. It all works fairly well.

I had the great idea of bushwalking local trails parks, with a field recorder (Zoom H6) and shotgun mic. It works using a battery bank, mobile hotspot for monitoring and all that good stuff.

The only issue I'm having is that the operating system, or more likely pulse audio isn't picking up the zoom H6 reliable. Unlike a basic interface, it needs to be booted up, and put into interface mode. I'm guessing that the Pi boots up, pulse starts and by the time I've managed to put the H6 into interface mode, it's sort of too late to be detected.

I've been hitting the sound card settings, manually specifying the H6 (from the aplay -l output), and checking the spectrogram/live audio. Sometimes that works. Most often however, I need to go back to sound settings and setting it to default for the H6 to detect. 

Am I doing this the wrong way? Should I just restart pulse audio, or the entire birdnet system once booted?

**Comment by mcguirepr89** [comment, marked as answer]
Hi, @dfoster81 -- Thanks for starting this discussion. For the record, I'm the only curmudgeon, everyone else is really nice!!! So you're in a good place!

I will tinker with this a bit and think on the best way to help, though I want to specify that you'll want to use the output from the `arecord -L` command to find the device name ALSA associates with the device (to bypass `pulseaudio`). That may be the only tweak you need for things to work as you need, but I'll keep thinking and perhaps someone else has had a similar experience and would want to weigh in.

**Reply by dfoster81** [reply]
Oh thanks so much, yep I just saw ticket #551 

This may be a Zoom specific thing, as the setup there was quite similar

**Reply by dfoster81** [reply]
Ok, so here's the relavent output of arecord -L
``plughw:CARD=H6,DEV=0
    H6, USB Audio
    Hardware device with all software conversions
sysdefault:CARD=H6
    H6, USB Audio
    Default Audio Device
front:CARD=H6,DEV=0
    H6, USB Audio
    Front output / input
dsnoop:CARD=H6,DEV=0
    H6, USB Audio
    Direct sample snooping device
usbstream:CARD=H6
    H6
    USB Stream Output

Looking at #551 you've recommended using the dsnoop value? It isn't taking dsnoop:CARD=H6,DEV=0, returning the error: 'Please Enter A Number"

**Reply by dfoster81** [reply]
My bad I was in the wrong field. You've absolutely nailed the issue thanks.

Might I humbly suggest including reference to arecord -L and dsnoop card names on the configuration page? :)

**Reply by mcguirepr89** [reply]
I'm very glad that was able to resolve your trouble.

> Might I humbly suggest including reference to arecord -L and dsnoop card names on the configuration page? :)

Good idea :+1: https://github.com/mcguirepr89/BirdNET-Pi/commit/69481061ea2677b3ddebf9c137d9534f08b762ff https://github.com/mcguirepr89/BirdNET-Pi/commit/5df0210df4c1d09ab2b19dcc36fa7bfcbba0f79c

Is that what you had in mind? I noticed that there was already a comment about how to set the sound card using `aplay -L` (which should be the same results as `arecord -L`) -- but I changed the `aplay` to the more apt `arecord` and added a line about opting for the `dnoop` device if it is available. Is that what you had in mind?

**Comment by dfoster81** [comment]
Yes that's amazing. My Linux sound architecture knowledge is very rusty, but I've seen at least one other issue on a Zoom based device with this stuff

. Hopefully it helps someone else :)

**Comment by alexbelgium** [comment]
> Elgato Wave XLR

Hi ! I see you are using a "Elgato Wave XLR" and had a question. I wanted to buy one but it says it is incompatible with Linux. Did you do something special to have it connect or did it work directly once you set the correct hardware path as described above?

Are clipguard and highpass filters active without the elgato softwares? Thanks!

Edit: I saw after posting that clipguard should stay enabled once it was as it is on the hardware side (https://www.reddit.com/r/elgato/comments/1d8re7o/what_is_clipguard_exactly/) - given the cost of the thing I'm still interested by your feedback if you're around!

**Comment by dfoster81** [comment]
Hey there. 

Been a while but it was just calling the device directly. DSP features such as clipgaurd etc do not work. 

You'd be better off with a Zoom or a Focusrite IMO. Something that can tie channels as a stereo pair if you ever want to go that way would be handy.

**Reply by alexbelgium** [reply]
Thanks very much for the fast response. I'll look at those. Birdnet really has an incredible community ;-)

**Comment by dfoster81** [comment]
I’ve used a few interfaces with it. Just my opinions on a fewZoom H6 great but overkill. The low pass filtering I’ve found to be great in the field. Have run it with a shotgun mic (Rode ntg3)M Audio mtrack solo. Would not recommend its a real noisy interface The Elgato. Kinda mid. It’s a waste on Linux as the dsp is all software based Zoom AMS 24 good basic interface recorded in stereo with it a bit. You can pair channels and match gains. I was using this with a pair of Rode M3s that have an inbuilt low cut filter Mackie profx3. 6 channel. Handy again with hardware low cut. have used   Rode m5s successfully. Never used focusrite but I’m pretty sure they have hardware lowcut filters. In terms of things like clip protection, you’ll learn how to set gains. Not super necessary. You’ll always have some kid in a hotted up VW golf blowing out your levels Good luck On 1 Oct 2024, at 6:38 PM, Alexandre ***@***.***> wrote:﻿
Thanks very much for the fast response. I'll look at those. Birdnet really has an incredible community ;-)

**Reply by alexbelgium** [reply]
Thanks for the detailed info, looks indeed you have been quite active in this field to have tested so many systems.

Based on your descriptions, I'll probably order several then to test : it seems Zoom has the H1 with 32 bits support comparable to the H6 (much cheaper), and probably the Focusrite. I'll look at the hardware low-cut too. My microphone will be a clippy EM272 XLR (I'm already using the trrs version that is already great but probably limited due to the lower voltage).

Quite excited to improve my system ;-) Since having birdnet-pi in place I've already cumulated 500k observations including many species I didn't expect in my neighbourhood, such as an Eurasian Eagle owl. I hope that improving the system will further allow to gain precision in detections and quality of recordings (fully aligned with the unexpected noises that can't be mitigated by any hardware : a surprising identification of Cuckoo was instead my neighboor calling for her cat).

**Reply by tphakala** [reply]
I have two BirdNET-Go (BirdNET-PI alternative) systems with Focusrite Scarlett interfaces one 2i2 Gen 3 and one Solo Gen 4. I think Scarlett is as good audio interface as anyone would need with realtime BirdNET analyzers, it has high gain (good mic amps) and works very well with Linux, only thing is that out of the box it does not remember phantom power setting (on) and I had to configure it as permanent setting on Windows, otherwise microphone lost its power any time RPi restarted etc.

I guess that I could have configured audio interface also using Linux but I don't have desktop installed and I was lazy to find out how to do it from Linux cli.

I also have one M-Audio M-Track solo, it also works well with Linux but it has significantly lower powered mic amps so audio it outputs is rather quiet. My microphones are DIY AOM5024 mics.

**Reply by alexbelgium** [reply]
Thanks for the feedback! Btw it's thanks to reading your posts that I decided to invest in the XLR microphone ;-) 

Edit : post edited, I was asking for your opinion on both but re-read your post and you actually recommend the Scarlett

**Reply by tphakala** [reply]
Clip safe was introduced with 4th Gen models but it is available only on models with 2 inputs or more, 2i2, 4i4 etc. Solo is missing it, but I haven't had any issues with clipping other than during very strong winds. I don't consider clip safe necessary at all, also low cut (high pass) filtering can be done quite easily using ALSA.

**Reply by alexbelgium** [reply]
thanks! and I saw you added high pass it in birdnet-go too ;-)
